client/ayon_resolve/plugins/create/create_editorial_package.py

```diff
@@ -1,88 +1,120 @@
 import json
 from copy import deepcopy
-from ayon_core.tools.context_dialog.window import (
-    ContextDialog,
-    ContextDialogController
-)
-from ayon_core.pipeline import get_current_project_name
-from ayon_api import get_folder_by_id, get_task_by_id, get_folder_by_path
-from ayon_core.pipeline.create.legacy_create import LegacyCreator
 
-from ayon_resolve.api import lib
-from ayon_resolve.api.plugin import get_editorial_publish_data
+from ayon_core.pipeline.create import CreatorError, CreatedInstance
 
+from ayon_resolve.api import lib, constants
+from ayon_resolve.api.plugin import ResolveCreator, get_editorial_publish_data
 
-class CreateEditorialPackage(LegacyCreator):
+
+class CreateEditorialPackage(ResolveCreator):
     """Create Editorial Package."""
 
-    name = "editorial_pkg"
+    identifier = "io.ayon.creators.resolve.editorial_pkg"
+    product_name = "editorial_pkgMain"
```

Comment:
A product name should technically not be hardcoded or overridden like this - but should adhere to the anatomy templates.

In this case you wouldn't need this at all and this product name appears to be the 'default' for the default anatomy template anyway. So I'd say, remove this.

(By default it uses the `Creator.get_product_name` implementation, if you're curious)

Reply:
Thanks for this suggestion, I didn't know about this method. 
Hopefully I've used it correctly, please let me know if not.

Reply:
Ah this one is on me - because I missed one bit which seeing it now... made me understand why you're doing this.
You can remove the manual calling of the method - because it's done for you. See [this comment](https://github.com/ynput/ayon-resolve/pull/36/files#r1790937622).

The `Creator.create` method's first argument is the `product_name`. Which it gets passed from `CreateContext.create` here: https://github.com/ynput/ayon-core/blob/8af66a0f783a959ca477418618775fe9081f6574/client/ayon_core/pipeline/create/context.py#L1153-L1171